demo/3000-home/next.config.js

```diff
@@ -1,20 +1,30 @@
 const NextFederationPlugin = require('@module-federation/nextjs-mf');
-
+const {promiseTemplate, promiseFactory} = require('@module-federation/nextjs-mf/lib/build-utils');
 module.exports = {
   webpack(config, options) {
     const { isServer } = options;
     if (!options.isServer) {
+      const remotes = {
+        shop: `shop@http://localhost:3001/_next/static/${
+          isServer ? 'ssr' : 'chunks'
+        }/remoteEntry.js`,
+        checkout: `checkout@http://localhost:3002/_next/static/${
+          isServer ? 'ssr' : 'chunks'
+        }/remoteEntry.js`,
+      }
       config.plugins.push(
         new NextFederationPlugin({
           name: 'home_app',
           filename: 'static/chunks/remoteEntry.js',
           remotes: {
-            shop: `shop@http://localhost:3001/_next/static/${
-              isServer ? 'ssr' : 'chunks'
-            }/remoteEntry.js`,
-            checkout: `checkout@http://localhost:3002/_next/static/${
-              isServer ? 'ssr' : 'chunks'
-            }/remoteEntry.js`,
+            shop: promiseTemplate(remotes.shop, promiseFactory(function(resolve,reject){
+              console.log('runing other promise');
+              setTimeout(() => {
```

Comment:
It will be very convenient for end users if we move `promiseFactory` inside `promiseTemplate`:

```js
promiseTemplate(remotes.shop, function(resolve,reject){...});
```

Reply:
hmmm, good point. I think its possible to do so